### PR TITLE
GH-539 CONV_P - Adds publish params for already existing functions

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @drealeed @dtsnell4 @jbrundage 

Please Review (rough copy with commented out code to show potential changes as well)

--

Adds publish params for already existing functions

Fixes GH-539

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/CONV_P/demos/dermatology.html

File: src/chart/Pie.js [hunk 1]
```diff
@@ -32,6 +32,9 @@
     Pie.prototype.implements(I2DChart.prototype);
 
     Pie.prototype.publish("paletteID", "default", "set", "Palette ID", Pie.prototype._palette.switch(),{tags:['Basic','Shared']});
+    Pie.prototype.publish("outerText", false, "boolean", "Sets label position inside or outside chart",null,{tags:['Basic']});
+    Pie.prototype.publish("radius", 100, "number", "Sets radius of pie chart",null,{tags:['Basic']});
```
Comment (jbrundage) [inline src/chart/Pie.js]: Specify what 100 means in this context. Is it pixels? Percentage of width?

File: src/chart/Pie.js [hunk 2]
```diff
@@ -32,6 +32,9 @@
...
+    Pie.prototype.publish("innerRadius", 0, "number", "Sets radius of center hole in pie chart (donut)",null,{tags:['Basic']});
```
Comment (jbrundage) [inline src/chart/Pie.js]: Specify what 0 means in this context. Is it pixels? Percentage of width?

File: src/chart/Pie.js [hunk 3]
```diff
@@ -10,9 +10,9 @@
         SVGWidget.call(this);
         I2DChart.call(this);
 
-        this._outerText = false;  //  Put label inside pie or outside (true/false)
-        this._radius = 100;       // px
-        this._innerRadius = 0;    // px
+        //this._outerText = false;  //  Put label inside pie or outside (true/false)
+        //this._radius = 100;       // px
+        //this._innerRadius = 0;    // px
```
Comment (GordonSmith) [inline src/chart/Pie.js]: remove commented code

File: src/common/Surface.js [hunk 4]
```diff
@@ -9,7 +9,7 @@
     function Surface() {
         SVGWidget.call(this);
 
-        this._menuPadding = 2;
+        //this._menuPadding = 2;
```
Comment (GordonSmith) [inline src/common/Surface.js]: remove commented code

File: src/common/Surface.js [hunk 5]
```diff
@@ -40,7 +40,7 @@
             }
         };
 
-        this._showContent = true;
+        //this._showContent = true;
```
Comment (GordonSmith) [inline src/common/Surface.js]: Remove commented code.

File: src/common/Surface.js [hunk 6]
```diff
@@ -59,6 +59,11 @@
     Surface.prototype.publish("buttonAnnotations", [], "array", "Button Array",null,{tags:["Intermediate"]});
     Surface.prototype.publish("buttonGutter", 25, "number", "Space Between Menu and Buttons",null,{tags:["Intermediate"]});
 
+    Surface.prototype.publish("showContent", true, "boolean", "Show Content",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menu", [], "array", "Menu List Data",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menuPadding", 2, "number", "Menu Padding",null,{tags:["Advance"]});
+
+
```
Comment (GordonSmith) [inline src/common/Surface.js]: remove blank line

File: src/map/Choropleth.js [hunk 7]
```diff
@@ -19,6 +19,7 @@
     Choropleth.prototype.implements(IChoropleth.prototype);
 
     Choropleth.prototype.publish("paletteID", "YlOrRd", "set", "Palette ID", Choropleth.prototype._palette.switch(),{tags:['Basic','Shared']});
+    Choropleth.prototype.publish("projection", "albersUsaPr", "set", "Projection Type", ["albersUsaPr", "orthographic", "mercator"]);
```
Comment (GordonSmith) [inline src/map/Choropleth.js]: Once a property is published the "overriding" method needs to call to the base.

File: src/common/Surface.js [hunk 8]
```diff
@@ -59,6 +59,11 @@
     Surface.prototype.publish("buttonAnnotations", [], "array", "Button Array",null,{tags:["Intermediate"]});
     Surface.prototype.publish("buttonGutter", 25, "number", "Space Between Menu and Buttons",null,{tags:["Intermediate"]});
 
+    Surface.prototype.publish("showContent", true, "boolean", "Show Content",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menu", [], "array", "Menu List Data",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menuPadding", 2, "number", "Menu Padding",null,{tags:["Advance"]});
```
Comment (GordonSmith) [inline src/common/Surface.js]: Once properties are published the "overriding" methods need to call to "new" corresponding base class methods.

File: src/common/Shape.js [hunk 9]
```diff
@@ -17,6 +17,7 @@
     Shape.prototype.publish("height", 24, "number", "Height",null,{tags:['Private']});
     Shape.prototype.publish("colorStroke", null, "html-color", "Stroke Color", null, {tags:['Private']});
     Shape.prototype.publish("colorFill", null, "html-color", "Fill Color", null, {tags:['Private']});
+    Shape.prototype.publish("radius", null, "number", "Radius",null,{tags:['Private']});
```
Comment (GordonSmith) [inline src/common/Shape.js]: Once properties are published the "overriding" methods need to call to "new" corresponding base class methods

File: src/chart/Pie.js (hunk 2 above)
Comment (GordonSmith) [inline src/chart/Pie.js]: Once properties are published the "overriding" methods need to call to "new" corresponding base class methods

File: src/other/Audio.js [hunk 10]
```diff
@@ -11,12 +11,14 @@
...
         this._tag = "audio";
 
-        this._source = [];
+        //this._source = [];
         this._sections = {};
     }
     Audio.prototype = Object.create(HTMLWidget.prototype);
     Audio.prototype._class += " other_Audio";
 
+    Audio.prototype.publish("source", [], "array", "Audio Source");
```
Comment (GordonSmith) [inline src/other/Audio.js]: Once properties are published the "overriding" methods need to call to "new" corresponding base class methods

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review

File: src/common/Surface.js [hunk 11]
```diff
@@ -53,31 +50,34 @@
...
     Surface.prototype.publish("content", null, "widget", "Content",null,{tags:["Private"]});
 
     Surface.prototype.publish("buttonAnnotations", [], "array", "Button Array",null,{tags:["Intermediate"]});
     Surface.prototype.publish("buttonGutter", 25, "number", "Space Between Menu and Buttons",null,{tags:["Intermediate"]});
 
+    Surface.prototype.publish("showContent", true, "boolean", "Show Content",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menu", [], "array", "Menu List Data",null,{tags:["Intermediate"]});
+    Surface.prototype.publish("menuPadding", 2, "number", "Menu Padding",null,{tags:["Advance"]});
+
+    Surface.prototype._menu = Surface.prototype.menu;
```
Comment (GordonSmith) [inline src/common/Surface.js]: _menu is being used already, must be causing havoc?

Comment (mlzummo) [inline src/common/Surface.js]: actually it still all worked somehow ... but im goin to change it

**Comment by GordonSmith** [comment]
Can you fix the commit message while your doing that?

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review

**Comment by GordonSmith** [comment]
@mlzummo - first two changes I tested failed (and I stopped reviewing at that point):  
Pie fails to load - http://rawgit.com/mlzummo/Visualization/CONV_P/demos/dermatology.html?src/chart/Pie
MultiChartSurface - "allowResize" is now broken.

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review fixed issues

**Comment by GordonSmith** [comment]
Closing PR as the very first change I tested is broken - "Pie.outerText(true)" You can reopen after you have tested your changes.

**Pull request closed without merging**